src/Ledger/Conway/Specification/Ledger/Properties/Base.lagda.md

```diff
@@ -113,6 +113,88 @@ module LEDGER-PROPS (tx : Tx) (Γ : LEnv) (s : LState) where
     dpMap-rmOrphanDRepVotes : ∀ certState govSt → dpMap (rmOrphanDRepVotes certState govSt) ≡ dpMap govSt
     dpMap-rmOrphanDRepVotes certState govSt = sym (fmap-∘ govSt) -- map proj₁ ∘ map (map₂ _) ≡ map (proj₁ ∘ map₂ _) ≡ map proj₁
 
+-- TODO: Move these proofs to agda-sets
+module _ {A V : Set} ⦃ mon : CommutativeMonoid 0ℓ 0ℓ V ⦄ ⦃ dA : DecEq A ⦄ {m m' : A ⇀ V} where
+
+  open import Relation.Binary.Bundles using (Poset)
+
+  rhs-∪ˡ : A ⇀ V
+  rhs-∪ˡ = filterᵐ? (sp-∘ (sp-¬ (∈-sp {X = dom (m ˢ)})) proj₁) m'
+
+  dom∪ˡˡ : dom (m ˢ) ⊆ dom ((m ∪ˡ m') ˢ)
+  dom∪ˡˡ = begin
+    dom (m ˢ)                   ⊆⟨ ∪-⊆ˡ ⟩
+    dom (m ˢ) ∪ dom (rhs-∪ˡ ˢ)  ≈⟨ ≡ᵉA.sym dom∪ ⟩
+    dom ((m ˢ) ∪ (rhs-∪ˡ ˢ))    ≈⟨ ≡ᵉA.refl ⟩
+    dom ((m ∪ˡ m') ˢ)
+    ∎
+    where
+      ⊆-Poset : Poset 0ℓ 0ℓ 0ℓ
+      ⊆-Poset = record
+        { Carrier = ℙ A
+        ; _≈_ = _≡ᵉ_
+        ; _≤_ = _⊆_
+        ; isPartialOrder = ⊆-PartialOrder
+        }
+      open import Relation.Binary.Reasoning.PartialOrder ⊆-Poset
+      open import Relation.Binary.Structures using (IsEquivalence)
+      module ≡ᵉA = IsEquivalence (≡ᵉ-isEquivalence {A = A})
+      open import Relation.Binary.Reasoning.Syntax
+      open ⊆-syntax _IsRelatedTo_ _IsRelatedTo_ ≤-go public
+
+  _∎→ : ∀ (A : Type) → A → A
+  A ∎→ = id
+  infix 3 _∎→
+
+  _→⟨_⟩_ : ∀ (x : Type) {y z : Type} → (x → y) → (y → z) → x → z
+  _ →⟨ f ⟩ g = g ∘ f
+  infixr 2 _→⟨_⟩_
+
+  begin→ : ∀ {A B : Type} → A → (A → B) → B
+  begin→ a f = f a
```

Comment:
These are used in the proofs below. Do they belong here or in some more general place?